### PR TITLE
Merge from functions branch to develop

### PR DESCRIPTION
**Pull request by Revarine**

### CODE REVIEW

File: calculator/calculator/calculator/Views/AddFunction.xaml.cs [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
+            }
+            else
+            {
+                MessageBox.Show("Параметр не может быть пустым.");
+            }
+        }
+
+        private void CreateNewFunctionClick(object sender, RoutedEventArgs e)
+        {
+            string functionName = FunctionNameTextBox.Text;
```
Comment (mebry) [inline calculator/calculator/calculator/Views/AddFunction.xaml.cs]: Why var is not used. Other places in the application use var. Fix it everywhere

File: calculator/calculator/calculator/Views/AddFunction.xaml.cs [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
+            string functionExpression = FunctionExpressionTextBox.Text;
+            bool isValid = Regex.IsMatch(functionName, @"^[a-zA-Z][a-zA-Z0-9]*$");
+            if (!isValid)
+            {
+                MessageBox.Show("В названии функции могуть быть только буквы или цифры, начинаться оно должно с буквы.");
```
Comment (mebry) [inline calculator/calculator/calculator/Views/AddFunction.xaml.cs]: In English

**Review by mebry — CHANGES_REQUESTED** [review]

File: calculator/calculator/calculator/Views/AddFunction.xaml.cs (hunk 1 above)
Comment (Revarine) [reply calculator/calculator/calculator/Views/AddFunction.xaml.cs]: Fixed in last commit ✅

File: calculator/calculator/calculator/Views/AddFunction.xaml.cs (hunk 2 above)
Comment (Revarine) [reply calculator/calculator/calculator/Views/AddFunction.xaml.cs]: Fixed in last commit ✅

**Review by mebry — APPROVED** [review]

**Pull request merged**